### PR TITLE
Add dependencies for ROM_EXT signer and build it by default

### PR DESCRIPTION
**Pull request by alphan**
This change adds the following dependencies required to build ROM_EXT signer to CI
configuration files, Docker container, and documentation:
- Cmake
- Go
- Perl
- Rust

Cmake, go, and perl are required to build BoringSSL, which is vendored by [mundane](https://github.com/google/mundane) (the rust library that the signer uses) and version numbers are from https://github.com/google/mundane#readme.

Rustup is the recommended way to install rust, which can be downloaded using the rustup-init.sh script. This change vendors rustup-init.sh so that we can directly run it instead of [piping the output of curl to sh](https://www.rust-lang.org/learn/get-started). We can then download the toolchain similar to `get-toolchain.py`.

I will announce this change before this PR is merged so that everyone is aware of new dependencies.

### CODE REVIEW

**Comment by imphil** [comment]
That's an interesting set of new dependencies :smile: Can you add them to https://docs.opentitan.org/doc/ug/install_instructions/? 

And since I know this to be a problem that will come up: can you please figure out which version of Go CentOS/RHEL7 ships, and how one would get a suitable version of go working on CentOS/RHEL6?

**Comment by alphan** [comment]
> And since I know this to be a problem that will come up: can you please figure out which version of Go CentOS/RHEL7 ships, and how one would get a suitable version of go working on CentOS/RHEL6?

Hmm, I'm not familiar with CentOS/RHEL packages at all, but it looks like it was [removed in 7.6](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/7.6_release_notes/chap-red_hat_enterprise_linux-7.6_release_notes-other_deprecated_functionality). From 7.6 release notes:
> Use the Go Toolset instead of golang
The golang package, previously available in the Optional channel, will no longer receive updates in Red Hat Enterprise Linux 7. Developers are encouraged to use the Go Toolset instead, which is available through the Red Hat Developer program.

All the examples that I came across require installing `epel-release` and `doc/ug/install_instructions/index.md` already mentions that the EPEL repository should be enabled so that's why I added `golang` to `yum-requirements.txt`.

Assuming that these ([7](https://epel.mirror.shastacoe.net/epel/7/x86_64/Packages/g/), [6](https://d2lzkl7pfhq30w.cloudfront.net/pub/archive/epel/6/x86_64/Packages/g/), links are from [fedora wiki](https://fedoraproject.org/wiki/EPEL#What_packages_and_versions_are_available_in_EPEL.3F)) are the correct places to look, packages for both versions seem to satisfy the minimum requirements. Can you also take a look?

Alternatively, folks can always follow the instructions at https://golang.org/ (already linked from `doc/ug/install_instructions/index.md`).

**Comment by imphil** [comment]
Sounds good, EPEL should do the trick then. No need to go further :-)

**Comment by imphil** [comment]
I'm a bit undecided, but I wonder if it would be better to install rust locally for the project, not for the user (e.g. into `$REPO_TOP/build/rust` or so). It doesn't matter for CI, but it would make things easier for users working on different projects. This approach would be similar to a Python venv. Can rustup do that, and we can then configure meson to use exactly this version of rust?

**Comment by alphan** [comment]
> I'm a bit undecided, but I wonder if it would be better to install rust locally for the project, not for the user (e.g. into `$REPO_TOP/build/rust` or so). It doesn't matter for CI, but it would make things easier for users working on different projects. This approach would be similar to a Python venv. Can rustup do that, and we can then configure meson to use exactly this version of rust?

We currently use `cargo` to build rust sources. I think the right way to address this is to add a `rust-toolchain.toml` file (please see: https://rust-lang.github.io/rustup/overrides.html). ATM, I'm not sure if we need it or if it will require any changes in `invoke_cargo.sh` (the wrapper script that is used to call cargo). @mcy, any thoughts?

File: ci/install-package-dependencies.sh [hunk 1]
```diff
@@ -134,3 +137,8 @@ tar -C /tools/verible -xf "$verible_tar" --strip-components=1
 
 # Propagate PATH changes to all subsequent steps of the job
 echo "##vso[task.setvariable variable=PATH]/tools/verible/bin:$PATH"
+
+# Install Rust
+sw/vendor/rustup/rustup-init.sh -y \
+    --default-toolchain "${RUST_VERSION}"
+export PATH="$HOME/.cargo/bin:$PATH"
```
Comment (imphil) [inline ci/install-package-dependencies.sh:144]: To make PATH available to subsequent build jobs in the pipeline you need to use the approach above `task.setvariable`.

File: doc/ug/install_instructions/index.md [hunk 2]
```diff
@@ -55,6 +55,10 @@ For packages listed below without a version number we have not determined a mini
 * libusb 1.0 (called libusbx in older distributions)
 * OpenOCD 0.11.0
 * [Verible](https://github.com/google/verible) {{< tool_version "verible" >}}
+* Cmake 2.8.11
+* Perl 5.6.1
+* [Go](https://golang.org/) 1.10
+* [Rust](https://www.rust-lang.org/) 1.52.1
```
Comment (imphil) [inline doc/ug/install_instructions/index.md:61]: I'd expect us to change the Rust version regularly; can you the version number into tool_requirements.py?

And then refer to it like this:

```suggestion
* [Rust](https://www.rust-lang.org/) {{< tool_version "rust" >}}
```

And this is currently only rust for x86_64? Maybe worth mentioning as well.

**Review by imphil — APPROVED** [review]
LGTM, thanks! Some small comments, but ready to go otherwise.

**Review by moidx — APPROVED** [review]

**Review by mcy — APPROVED** [review]

**Comment by alphan** [comment]
I tried adding the `rust-toolchain.toml` file but this didn't work because `cargo` looks for this file starting from the current working directory, which cannot be set in meson. We can update `invoke_cargo.sh` to `cd` to the source dir before running `cargo` but there seems to be additional things that could be improved/simplified in that script. Created #6549 to address this in a separate PR.

**Comment by martin-lueker** [comment]
So I can't speak for the EPEL path at all, but downloading and building golang and rustup went pretty smoothly.   There were a few kinks in getting our environment files together, but I think we are running again.  Thanks @alphan for all your help!

As I mentioned to @alphan in an e-mail it seems however that the mundane crate complained that we needed cmake 3.3 (even though I can see that the crates.io lists cmake 2.8.11 as the base dependency).

Thanks again for the announcement letting us know of this change.

**Comment by alphan** [comment]
@martin-lueker created google/mundane#31.

File: yum-requirements.txt [hunk 3]
```diff
@@ -13,11 +13,13 @@ autoconf
 bison
 build-essential
 clang-format
+cmake
```
Comment (sriyerg) [inline yum-requirements.txt:16]: I need to resurrect and finish off PR #5192. Some of these package names are invalid.

**Pull request merged**